logger.go

```diff
@@ -154,3 +174,12 @@ func (l *Logger) log(level Level, e *Entry, msg string) {
 		stdlog.Printf("error logging: %s", err)
 	}
 }
+
+// logl the message. Similar to log but uses a lazy logSupplier to
+// avoid creating the message if the level is not met.
+func (l *Logger) logl(level Level, e *Entry, msg logSupplier) {
+	if level < l.Level {
+		return
+	}
+	l.log(level, e, msg())
```

Comment:
Skips invoking msg() if level is not met otherwise creates the message and defer to the main logging flow.